synapse/models/files.py

```diff
@@ -121,62 +121,92 @@ def seedFileSha1(self, prop, valu, **props):
-                    ('xref:intval', {'ptype': 'int', 'ro': 1}),
-                    ('xref:strval', {'ptype': 'str', 'ro': 1}),
+                    ('file', {'ptype': 'file:bytes', 'doc': 'The guid of the file containing the image.', 'ro': 1}),
+                    ('xref', {'ptype': 'propvalu',
+                         'doc': 'The form=valu of the object referenced in the image, e.g., geo:place=<guid_of_place>.', 'ro': 1}),
+                    ('xref:prop', {'ptype': 'str', 'doc': 'The property (form) of the referenced object, as specified by the propvalu.', 'ro': 1}),
+                    ('xref:intval', {'ptype': 'int',
+                         'doc': 'The value of the property of the referenced object, as specified by the propvalu, if the value is an integer.', 'ro': 1}),
+                    ('xref:strval', {'ptype': 'str',
+                          'doc': 'The value of the property of the referenced object, as specified by the propvalu, if the value is a string.', 'ro': 1}),
                 ]),
 
                 ('file:txtref', {}, [
-                    ('file', {'ptype': 'file:bytes', 'ro': 1}),
-                    ('xref', {'ptype': 'propvalu', 'ro': 1}),
-                    ('xref:prop', {'ptype': 'str', 'ro': 1}),
-                    ('xref:intval', {'ptype': 'int', 'ro': 1}),
-                    ('xref:strval', {'ptype': 'str', 'ro': 1}),
+                    ('file', {'ptype': 'file:bytes', 'doc': 'The guid of the file containing the reference.', 'ro': 1}),
+                    ('xref', {'ptype': 'propvalu',
+                         'doc': 'The form=valu of the object referenced in the fiole, e.g., inet:fqdn=foo.com.', 'ro': 1}),
+                    ('xref:prop', {'ptype': 'str',
+                         'doc': 'The property (form) of the referenced object, as specified by the propvalu.', 'ro': 1}),
+                    ('xref:intval', {'ptype': 'int',
+                         'doc': 'The value of the property of the referenced object, as specified by the propvalu, if the value is an integer.', 'ro': 1}),
+                    ('xref:strval', {'ptype': 'str',
+                         'doc': 'The value of the property of the referenced object, as specified by the propvalu, if the value is a string.', 'ro': 1}),
                 ]),
 
                 ('file:path', {}, (
-                    ('dir', {'ptype': 'file:path', 'doc': 'The parent directory for this path.'}),
-                    ('ext', {'ptype': 'str:lwr', 'doc': 'The file extension ( if present ).'}),
-                    ('base',
-                     {'ptype': 'file:base', 'doc': 'The final path component, such as the filename, of this path.'}),
+                    ('dir', {'ptype': 'file:path',
+                         'doc': 'The parent directory of the file path. Can be the directory path (if the file:path ends in a file name) or the parent directory (if the file:path is itself a directory).', 'ro': 1}),
```

Comment:
This doc is confusing to me. A `file:path` is a path that can be to either a file or a directory.

Reply:
Changed - removed all but first sentence. Over-explaining.